### DISCUSSION TITLE
TypeError: '<' not supported between instances of 'NoneType' and 'int' on trying to acess http://localhost:8000/graphql

### DISCUSSION THREAD

**Discussion by krauss** [encode/starlette#1632] [Q&A]
Hey guys, I've been using Starlette with FastAPI and Strawberry to expose a GraphQL API for my web app. However, since last week I noticed I'm getting the error below when trying to acess http://localhost:8000/graphql:

```python
INFO:     127.0.0.1:45822 - "GET /graphql HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/venv/lib64/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 372, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/venv/lib64/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/routing.py", line 665, in __call__
    await route.handle(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/routing.py", line 261, in handle
    await self.app(scope, receive, send)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./strawberry/asgi/__init__.py", line 47, in __call__
    await self.http_handler_class(
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./strawberry/asgi/handlers/http_handler.py", line 42, in handle
    sub_response = Response(
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/responses.py", line 55, in __init__
    self.init_headers(headers)
  File "/home/jrkrauss/OpenSource/Workspaces/python/ceprol-back/./starlette/responses.py", line 85, in init_headers
    and not (self.status_code < 200 or self.status_code in (204, 304))
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```
These are some of the libraries I'm using:
fastapi==0.77.0
starlette==0.20.0
strawberry==0.111.2
graphql-core==3.2.1
uvicorn==0.17.6
uvloop==0.16.0

After debugging the libraries, I noticed that the chunk of code below, taken from file [response.py, line 84](https://github.com/encode/starlette/blob/e7a92ee4cd43b6d7acf4e2fdedb20fe0e86f9620/starlette/responses.py#L84) is causing the error :
```python
body = getattr(self, "body", None)
if (
    body is not None
    and populate_content_length
    and not (self.status_code < 200 or self.status_code in (204, 304))
):
```

To summarize, the error above happens when `self.status_code is None`. This is how to fix it:

```python
body = getattr(self, "body", None)
if (
    body is not None
    and self.status_code is not None
    and populate_content_length
    and not (self.status_code < 200 or self.status_code in (204, 304))
):
```
Should I open a PR to fix it or is this something that's already on your radar?

Cheers!

**Comment by snosratiershad** [comment]
I will just mention [this issue](https://github.com/strawberry-graphql/strawberry/issues/1593) and [this PR](https://github.com/strawberry-graphql/strawberry/pull/1594). looks like it's not Starlette.

**Comment by mahyarmirrashed** [comment]
@krauss Thanks for the solution. For those of us frozen to this particular version, your solution has helped!
